include/aspect/postprocess/particle_pdf.h

```diff
@@ -0,0 +1,137 @@
+/*
+  Copyright (C) 2025 - by the authors of the ASPECT code.
+
+  This file is part of ASPECT.
+
+  ASPECT is free software; you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation; either version 2, or (at your option)
+  any later version.
+
+  ASPECT is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with ASPECT; see the file LICENSE.  If not see
+  <http://www.gnu.org/licenses/>.
+*/
+
+#ifndef _aspect_postprocess_particle_pdf_h
+#define _aspect_postprocess_particle_pdf_h
+
+#include <aspect/postprocess/interface.h>
+#include <aspect/simulator_access.h>
+#include <deal.II/particles/particle_iterator.h>
+#include <algorithm>
+#include <limits>
+#include <deal.II/base/table.h>
+
+namespace aspect
+{
+  namespace Postprocess
+  {
+    /**
+     * This class handles the point-density function for describing the clustering
+     * of particles.
+     */
+    template <int dim>
+    class ParticlePDF
+    {
+      public:
+
+        /**
+         * This is the constructor for ParticlePDF. The class holds
+         * the values of the point-density function at every point at which
+         * the function is defined. It also computes statistical values
+         * from the point-density function including the mean and standard deviation.
```

Comment:
Starting with `The class holds`, this text should actually go into the documentation of the class itself.